lib/rules/newline-after-var.js

```diff
@@ -146,7 +152,7 @@ module.exports = {
          */
         function checkForBlankLine(node) {
             const lastToken = getLastToken(node),
-                nextToken = sourceCode.getTokenAfter(node),
+                nextToken = lastToken === sourceCode.getLastToken(node) ? sourceCode.getTokenAfter(node) : sourceCode.getLastToken(node),
```

Comment:
This is necessary because otherwise the behavior is incorrect in places like this:

```js
var foo = 1
;
(bar).baz();
```

I added another commit to explain the reasoning in a comment, and also added tests where the original line fails.